### PR TITLE
add update_quality command

### PR DESCRIPTION
**Pull request by pnadolny13**
- new command for updating all quality attributes based on new inputs
- uses https://docs.meltano.com/cloud/connectors#connector-quality-matrix as a guide but has some slightly different opinions when it requires judgement
- tests to assert the expectations. I included every variation we had.
- this reads off the hub metrics file provided by Squared
- currently responsiveness is hard coded but we can add it to the hub metrics file
- partners are only matatika/autoidm/hotglue as of now

Other:
- I had to update the github action to clone the hub and pass in the HUB_ROOT_PATH for the tests to pass

@tayloramurphy let me know if you disagree with any of the test cases. Its trivial to update them and refresh https://github.com/meltano/hub/pull/1347.

### CODE REVIEW

File: tests/test_meltano_utils.py [hunk 1]
```diff
@@ -406,3 +406,71 @@ def test_sdk_about_parsing_skip_default_dates():
...
+        [["meltanolabs", True, 6, "high"], "gold"],
+        [["meltanolabs", False, 0, "low"], "gold"],
+        [["meltanolabs", False, 0, "medium"], "gold"],
+        [["meltanolabs", False, 0, "high"], "gold"],
+        [["meltanolabs", False, 1, "low"], "gold"],
+        [["meltanolabs", False, 1, "medium"], "gold"],
+        [["meltanolabs", False, 1, "high"], "gold"],
+        [["meltanolabs", False, 6, "low"], "gold"],
+        [["meltanolabs", False, 6, "medium"], "gold"],
+        [["meltanolabs", False, 6, "high"], "gold"],
```
Comment (tayloramurphy) [inline tests/test_meltano_utils.py:432]: @pnadolny13 I would say non-sdk based connectors should not be Gold. Perhaps max is silver and it's a clear signal we need to remake it on the SDK.

The only exception would be a high-usage/responsiveness connector - but it depends on what the connector actually is.

File: tests/test_meltano_utils.py [hunk 2]
```diff
@@ -406,3 +406,71 @@ def test_sdk_about_parsing_skip_default_dates():
...
+        # Community
+        [["foobar", True, 0, "low"], "silver"],
+        [["foobar", True, 0, "medium"], "silver"],
+        [["foobar", True, 0, "high"], "silver"],
+        [["foobar", True, 1, "low"], "silver"],
+        [["foobar", True, 1, "medium"], "silver"],
+        [["foobar", True, 1, "high"], "silver"],
+        [["foobar", True, 6, "low"], "silver"],
+        [["foobar", True, 6, "medium"], "silver"],
+        [["foobar", True, 6, "high"], "silver"],
```
Comment (tayloramurphy) [inline tests/test_meltano_utils.py:463]: I would think these could be Gold. Just b/c it's a community connector doesn't mean it _can't_ be Gold. It's just not automatic.

**Review by tayloramurphy — APPROVED** [review]
Couple of comments, but looks great!

File: tests/test_meltano_utils.py (hunk 1 above)
Comment (pnadolny13) [reply tests/test_meltano_utils.py:432]: @tayloramurphy I agree with that, here are the meltano/meltanolabs variants that arent SDK based. I think silver/bronze are accurate for all of these:

- tap-salesforce - meltanolabs
- tap-sftp - meltanolabs
- tap-gitlab - meltanolabs
- target-athena - meltanolabs
- target-sqlite - meltanolabs
- tap-fastly - meltano (GitLab archived)

File: tests/test_meltano_utils.py (hunk 2 above)
Comment (pnadolny13) [reply tests/test_meltano_utils.py:463]: @tayloramurphy ah yes youre right.

File: tests/test_meltano_utils.py (hunk 1 above)
Comment (tayloramurphy) [reply tests/test_meltano_utils.py]: @pnadolny13 yep - that list feels like a silver/bronze list!

**Comment by pnadolny13** [comment]
@tayloramurphy I made the changes to reflect your feedback and also made singer-io and airbyte minimum of bronze. I'm going to take a deeper dive but this is the current state https://docs.google.com/spreadsheets/d/19u-W4Krze6Htm7ywPGWpHf3Rkj5SMZd85EVmDQHfQek/edit#gid=0.

**Review by tayloramurphy — APPROVED** [review]

**Pull request merged**